mithril-signer/src/runtime/state_machine.rs

```diff
@@ -584,210 +602,178 @@ mod tests {
             .await
             .expect("Cycling the state machine should not fail");
 
-        if let SignerState::Registered { epoch: _ } = state_machine.get_state().await {
+        if let SignerState::RegisteredNotAbleToSign { epoch: _ } = state_machine.get_state().await {
         } else {
             panic!(
-                "state machine did not return a RegisteredState but {:?}",
+                "state machine did not return a RegisteredNotAbleToSign state but {:?}",
                 state_machine.get_state().await
             );
         }
     }
 
     #[tokio::test]
-    async fn registered_to_unregistered() {
+    async fn unregistered_to_ready_to_sign() {
         let mut runner = MockSignerRunner::new();
+        runner.expect_upkeep().returning(|| Ok(())).once();
         runner
-            .expect_get_current_time_point()
+            .expect_get_epoch_settings()
             .once()
+            .returning(|| Ok(Some(fake_data::epoch_settings())));
+
+        runner
+            .expect_inform_epoch_settings()
+            .with(predicate::eq(fake_data::epoch_settings()))
+            .once()
+            .returning(|_| Ok(()));
+
+        runner
+            .expect_get_current_time_point()
+            .times(2)
             .returning(|| Ok(TimePoint::dummy()));
         runner
-            .expect_update_era_checker()
+            .expect_update_stake_distribution()
             .once()
-            .returning(|_e: Epoch| Ok(()));
+            .returning(|_| Ok(()));
+        runner
+            .expect_register_signer_to_aggregator()
+            .once()
+            .returning(|| Ok(()));
 
-        let state_machine = init_state_machine(SignerState::Registered { epoch: Epoch(0) }, runner);
+        runner
+            .expect_can_sign_current_epoch()
+            .once()
+            .returning(|| Ok(true));
+
+        let state_machine = init_state_machine(
+            SignerState::Unregistered {
+                epoch: TimePoint::dummy().epoch,
+            },
+            runner,
+        );
 
         state_machine
             .cycle()
             .await
             .expect("Cycling the state machine should not fail");
-        assert_eq!(
-            SignerState::Unregistered {
-                epoch: TimePoint::dummy().epoch
-            },
-            state_machine.get_state().await
-        );
+
+        if !state_machine.get_state().await.is_ready_to_sign() {
+            panic!(
+                "state machine did not return a ReadyToSign state but {:?}",
+                state_machine.get_state().await
+            );
+        }
```

Comment:
Maybe this is enough:
```suggestion
        assert!(state_machine.get_state().await.is_ready_to_sign());
```

Or use an `assert_eq!` on the expected state as it was done previously?